src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -675,6 +685,7 @@ void DecisionTree<FitnessFunction,
         numClasses,
         UseWeights ? weights.subvec(begin, begin + count - 1) : weights);
   }
+  return -bestGain;
```

Comment:
Do we need to negate a second time here?

Reply:
Yes, if as we can see we are maximizing entropy. Objective function always minimizes something. So returning negative entropy of final tree is reasonable.

Reply:
Sorry, I thought that it was avoidable since we're already negating `childGain` earlier, but I realize that the code you have here also handles the case where no split is made (in which case `bestGain` needs to be negated).